check_api/src/main/java/com/google/errorprone/scanner/ErrorProneScanner.java

```diff
@@ -185,6 +185,18 @@ public ErrorProneScanner(Iterable<BugChecker> checkers, Map<String, SeverityLeve
     ImmutableSet.Builder<Class<? extends Annotation>> annotationClassesBuilder =
         ImmutableSet.builder();
     for (BugChecker checker : this.bugCheckers) {
+      // When disabling a check using -Xep:<BugChecker>:OFF
+      // its `SeverityLevel` is removed from the `severityMap` in
+      // com.google.errorprone.scanner.ScannerSupplier#L210
+      //
+      // If a checker is disabled, there's no need to register the checker.
+      // If the checker is still registered, attempting to report any matches will result in a a
+      // NoSuchElementException while retrieving the
+      // severity level in:
+      // `com.google.errorprone.matchers.Description#severity()`.
```

Comment:
I like the level of detail for the explanation, but I feel it doesn't belong here. It should be in the PR as extra context maybe, to make the reviewing easier though. (It helped me as well 😄).

I pushed a proposal for a one-liner explanation. PTAL 😄.